File: atividade-22/atividade.go
package main

import (
	"fmt"
)

type AFD struct {
	state string
}

func NewAFD() *AFD {
	return &AFD{state: "q0"} 
}

func (afd *AFD) transition(c rune) {
	switch afd.state {
	case "q0":
		if c == 'a' {
			afd.state = "q1"
		} else if c == 'b' {
			afd.state = "q2"
		}
	case "q1":
		if c == 'a' {
			afd.state = "q2"
		} else if c == 'b' {
			afd.state = "q0"
		}
	case "q2":
		if c == 'a' {
			afd.state = "q0"
		} else if c == 'b' {
			afd.state = "q1"
		}
	}
}

func (afd *AFD) isAccepted(input string) bool {
	afd.state = "q0" 
	for _, c := range input {
		afd.transition(c)
	}
	return afd.state == "q0" 
}

func main() {
	afd := NewAFD()
	fmt.Println(afd.isAccepted("aaabbb") 
	fmt.Println(afd.isAccepted("aab"))    
	fmt.Println(afd.isAccepted("aabb"))   
	fmt.Println(afd.isAccepted("aaa"))    
	fmt.Println(afd.isAccepted("bbb"))    
	fmt.Println(afd.isAccepted("ababab")) 
}

File: atividade-22/atividade.py
class AFD:
    def __init__(self):
        self.state = 'q0'  

    def transition(self, char):
        if self.state == 'q0':
            if char == 'a':
                self.state = 'q1'
            elif char == 'b':
                self.state = 'q2'
        elif self.state == 'q1':
            if char == 'a':
                self.state = 'q2'
            elif char == 'b':
                self.state = 'q0'
        elif self.state == 'q2':
            if char == 'a':
                self.state = 'q0'
            elif char == 'b':
                self.state = 'q1'

    def is_accepted(self, string):
        self.state = 'q0'  
        for char in string:
            self.transition(char)
        return self.state == 'q0'  

if __name__ == '__main__':
    afd = AFD()
    print(afd.is_accepted("aaabbb")) 
    print(afd.is_accepted("aab"))     
    print(afd.is_accepted("aabb"))    
    print(afd.is_accepted("aaa"))    
    print(afd.is_accepted("bbbbbb"))     
    print(afd.is_accepted("ababab")) 